### PR TITLE
added mocks for test_geonames

### PR DESCRIPTION
**Pull request by ebreton**
Hi @DenisCarriere ,

I have created a branch to mock the tests with https://pypi.python.org/pypi/requests-mock, and tested the lib on test_geonames.py and it looks straightforward.

If you agree, let's work together on this branch, and let's use this pull request for discussion.

Please have a try with the existing tests and let me know what you think about it. Meanwhile, I will play a bit on the support for multiple results.

Cheers
Manu

### CODE REVIEW

File: tests/test_geonames.py [hunk 1]
```diff
@@ -2,46 +2,186 @@
...
-    assert g.country_code == 'CA'
-    assert g.state == 'Ontario'
-    assert g.state_code == '08'
-    assert g.description == 'capital of a political entity'
-    assert g.class_description == 'city, village,...'
-    assert g.feature_class == 'P'
-    assert g.code == 'PPLC'
-    assert g.geonames_id == 6094817
+    url = 'http://api.geonames.org/searchJSON?q=Ottawa%2C+Ontario&fuzzy=0.8&username=mock&maxRows=1'
+    text = """{
```
Comment (DenisCarriere) [inline tests/test_geonames.py]: We could simply store these results as JSON, for example we could have all the `mock` data stored in `tests/data`.

I've done a similar thing with my Javascript geocoder (unfinished).

https://github.com/DenisCarriere/geocoder-geojson/tree/master/test/in

**Comment by ebreton** [comment]
Good suggestion for test data. Applied.

I have also added a few mocks and fix the issue that was breaking the build for Google.

Oddly enough, I do not succeed in mocking all google calls... 
- google_reverse does not recognize the mocked url
- google_timezone does not recognize the json data

I think we reach a state where you can merge the PR :)

Il the build pass on master, I reckon you could also push a new version on pypi :)

**Comment by DenisCarriere** [comment]
👍 Awesome

> Il the build passes on master, I reckon you could also push a new version on pypi :)

You've been pushing quite a few commits, let me know whenever you're ready for a new version on PyPi.

**Comment by ebreton** [comment]
he he :)

and the build passes now 👍 

you can push to pypi, indeed ! I will stop for this week and add a bit of documentation on next week-end I think

**Comment by DenisCarriere** [comment]
haha awesome! 👍 I'll publish a new PyPi version

**Comment by DenisCarriere** [comment]
Ok i've published `v1.24.0`, I [updated the Makefile](https://github.com/DenisCarriere/geocoder/blob/master/Makefile#L23-L25) with the appropriate commands.

```bash
$ python setup.py sdist upload -r pypi
$ python setup.py bdist_wheel upload -r pypi
```

Also you need to add your PyPi credentials `~/.pypirc`.

```yml
[distutils]
index-servers =
  pypi

[pypi]
repository: https://upload.pypi.org/legacy/
username: <USERNAME>
password: <PASSWORD>
```

**Pull request merged**